### PR TITLE
Added content hash to json export

### PR DESCRIPTION
**Pull request by bavovanachte**

### CODE REVIEW

**Comment by bavovanachte** [comment]
PR in the context of #31 : This adds a hash of the content of the item to the json file.

**Comment by bavovanachte** [comment]
Bandit seems to flag a security issue for the use of MD5. Given that it's not exactly a security-critical application, I don't see it as a valid issue. i see 2 ways of fixing it:

- Disable this check completely in bandit (I don't believe anything is security-critical?). I've checked quickly and it seems that disabling a warning only for a given file doesn't seem possible (could be wrong)
- Switch to a more secure SHA algorithm. I've picked MD5 as it is the fastest (and sphinx doc building can already take some time), but I'm not exactly sure what the impact is of switching to something else.

**Comment by SteinHeselmans** [comment]
we can disable the check

**Comment by SteinHeselmans** [comment]
can we add a test somehow for this
- unit test: 2 same contents produce same hash
- unit test: export of item with no content gives the predefined hash value 0

**Comment by Letme** [comment]
I triaged the codeclinate issue. I think it would be great to print a warning if we already have the matching hash as suggested by @SteinHeselmans

**Comment by SteinHeselmans** [comment]
Not entirely @Letme , the hash is not designed to identify a doc-item, we have the item-id for that which needs to be unique. Different items in the container can have the same hash.
The hash came to live in order to spot if the content of a doc-item has been changed over time. So we need to unit test whether two different items with same text-content produce the same hash (see above comment).

Special case is an item which doesn't have any text-content, it should also produce a hash. And the value of that hash is pre-defined to be 0 in the implementation.

**Comment by bavovanachte** [comment]
Good idea on the rests. I'll have some time this week to work on them

File: mlx/traceable_base_class.py [hunk 1]
```diff
@@ -183,6 +184,10 @@ def to_dict(self):
             data['caption'] = caption
         data['document'] = self.docname
         data['line'] = self.lineno
+        if self.content:
+            data['content-hash'] = hashlib.md5(self.content.encode('utf-8')).hexdigest()
+        else:
+            data['content-hash'] = 0
```
[the remark this answers is not in the document]
Comment (bavovanachte) [reply mlx/traceable_base_class.py]: That makes sense. Will do

[the remark this answers is not in the document]
Comment (bavovanachte) [reply mlx/traceable_base_class.py]: See a146dc5e5bda08a344381d917244abd9655861bf

**Comment by bavovanachte** [comment]
Added tests in a146dc5 and dcb33e3

@SteinHeselmans is this what you had in mind? I didn't add extra test cases, just added checks to the existing ones.

File: tests/test_traceable_item.py [hunk 2]
```diff
@@ -77,6 +77,8 @@ def test_set_caption(self):
         self.assertEqual(self.docname, data['document'])
         self.assertEqual(0, data['line'])
         self.assertEqual({}, data['targets'])
+        print(data['content-hash'])
```
Comment (SteinHeselmans) [inline tests/test_traceable_item.py]: debug leftover?

Comment (bavovanachte) [reply tests/test_traceable_item.py]: Better now!

**Pull request merged**